File: api/src/main/java/com/masterson/api/model/Candle.java
package com.masterson.api.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class Candle {
    private float close;
    private String closeTime;
    private float high;
    private Interval interval;
    private float low;
    private float open;
    private String openTime;
    private Source source;
    private String ticker;
}

File: web/src/types/Candle.ts
import {Granularity} from './enums/Granularity';
import {Signal} from './enums/Signal';
import {IndicatorEntry} from './IndicatorEntry';

export interface Candle {
    close: number;
    granularity: Granularity;
    high: number;
    indicators: IndicatorEntry[];
    low: number;
    open: number;
    signals: Signal[];
    ticker: string;
    timestamp: string;
    volume: number;
}
